hackathon/models.py

```diff
@@ -0,0 +1,177 @@
+
+# When it seemed relevant, I moved the ForeignKey field to another model and
+# used the one listed in the schema as the related_name. I may have been
+# erroneous to do so but it can be corrected easily.
+# My reasoning is explained in comments above the ForeignKey fields.
+
+
+# Create your models here.
+class Hackathon(models.Model):
+    """Model representing a Hackathon. It is connected by a foreign key to 
+    User, HackAwards and HackTeam. Optional Fields: judges, organiser.
+    "awards" and "teams" are related tables. They have been moved to
+    HackAwardCategory and HackTeam respectively. Please see comments there."""
+    created = models.DateTimeField(default=timezone.now)
+    updated = models.DateTimeField(auto_now=True)
+    # Each model can only be created by one user: One To Many
+    created_by = models.ForeignKey(User,
+                                   on_delete=models.CASCADE,
+                                   related_name="hackathon_created_by")
+    display_name = models.CharField(default="", max_length=254)
+    description = models.TextField()
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
+    # Hackathons can have numerous judges and
+    # users could be the judges of more than one Hackathon: Many to Many
+    judges = models.ManyToManyField(User,
+                                    blank=True,
+                                    related_name='hackathon_judges')
+    # One organiser could organise more than one Hackathon: One To Many
+    organiser = models.ForeignKey(User,
+                                  null=True,
+                                  blank=True,
+                                  on_delete=models.SET_NULL,
+                                  related_name="hackathon_organiser")
+
+    def __str__(self):
+        return self.display_name
+
+
+class HackAwardCategory(models.Model):
+    """Model representing a HackAwardCategory. It is connected by a foreign key to 
+    User, Hackathon and HackProject. Optional fields: winning_project."""
+    created = models.DateTimeField(default=timezone.now)
+    updated = models.DateTimeField(auto_now=True)
+    # Each model can only be created by one user: One To Many
+    created_by = models.ForeignKey(User,
+                                   on_delete=models.CASCADE,
+                                   related_name="hackawardcategory_created_by")
+    display_name = models.CharField(default="", max_length=254)
+    description = models.TextField()
+    # a Category will only apply to one Hackathon and 
+    # a Hackathon has numerous categories: One to Many.
+    # If the category was going to be reused, instead, use Many to Many.
+    hackathon = models.ForeignKey("Hackathon",
+                                  on_delete=models.CASCADE,
+                                  related_name="awards")
+    # One category can have one winner: One to One
+    winning_project = models.OneToOneField("HackProject",
```

Comment:
You use quotes to reference an object which has not yet been defined. From here:
https://docs.djangoproject.com/en/3.1/ref/models/fields/#foreignkey

> If you need to create a relationship on a model that has not yet been defined, you can use the name of the model, rather than the model object itself

Reply:
Yep. Let's leave that then.